### PR TITLE
Make close condition more readable

### PR DESCRIPTION
**Pull request by rgs1**
This is a follow-up to #6942. This makes it a bit more readable. We
spent some time debugging this too, so it would be nice to ensure
it's readable next time we bump into something around this code
surface :-)

Signed-off-by: Raul Gutierrez Segales <rgs@pinterest.com>

### CODE REVIEW

File: source/common/upstream/health_checker_impl.cc [hunk 1]
```diff
@@ -270,20 +270,43 @@ void HttpHealthCheckerImpl::HttpActiveHealthCheckSession::onResponseComplete() {
 
   // It is possible for this session to have been deferred destroyed inline in handleFailure()
   // above so make sure we still have a connection that we might need to close.
```
Comment (mattklein123) [inline source/common/upstream/health_checker_impl.cc]: Can you move this comment down into the function about the client null check?

Comment (rgs1) [reply source/common/upstream/health_checker_impl.cc]: Yup, done.

**Comment by rgs1** [comment]
Yeah I'll ensure we have full coverage too.

**Comment by rgs1** [comment]
Coverage seems to be already OK, per https://217235-65214191-gh.circle-artifacts.com/0/coverage/coverage.source_common_upstream_health_checker_impl.cc.html

**Pull request merged**